iree/base/flatbuffer_util.h

```diff
@@ -33,6 +33,42 @@
 
 namespace iree {
 
+// A helper wrapper that moves the wrapped object on copy.
+// This is particularly handy for capturing unique_ptrs in lambdas.
+// Usage example:
+//
+//  std::unique_ptr<Foo> foo_ptr(new Foo());
+//  move_on_copy<std::unique_ptr<Foo>> foo(std::move(foo_ptr));
+//  auto some_lambda = [bar]() { ... }
+//
+template <typename T>
+struct move_on_copy {
+  explicit move_on_copy(T&& t) : value(std::move(t)) {}
+
+  move_on_copy(move_on_copy const& other) : value(std::move(other.value)) {}
+
+  move_on_copy(move_on_copy&& other) : value(std::move(other.value)) {}
+
+  move_on_copy& operator=(move_on_copy const& other) {
+    value = std::move(other.value);
+    return *this;
+  }
+
+  move_on_copy& operator=(move_on_copy&& other) {
+    value = std::move(other.value);
+    return *this;
+  }
+
+  mutable T value;
+};
+
+// Utility to aid in moving ref_ptr's into closures.
+//
+// Usage:
+//   auto baton = MoveToLambda(my_ref);
+//   DoSomething([baton] () { baton.value; });
+#define IreeMoveToLambda(p) ::iree::move_on_copy<decltype(p)>(std::move(p))
```

Comment:
Why move these from `iree/base/memory.h`? These changes are from the "Removing/moving some rarely used utilities." commit, can you elaborate? memory.h seemed more descriptive to me, and these aren't really flatbuffer-specific (even if their only uses are there right now).

Reply:
I'm trying to delete as much code as possible, and one stage of that is de-utilifying things that are used in single places. Here, this is only needed by this file (which itself isn't really needed and will be moved to iree/testing/ or deleted in the future), and to prevent new deps from arising it's useful to move it more local to its usage sooner rather than later.